pages/admin/balance.tsx

```diff
@@ -86,6 +96,7 @@ function AdminBalancePage(props) {
 
   React.useEffect(() => {
     const run = async () => {
+      getFilAddress(state, setState, props.api);
```

Comment:
Do you want one to execute before the other? I feel like the `setState` race condition here will be a little gross..

I would structure the run method as follows:

```
  React.useEffect(() => {
    const run = async () => {
      let newState = { account: null, amount: 0, loading: true };
      newState = await getFilAddress(newState, props.api);
      newState = await getBalance(newState, props.api);
      setState({ ...state, newState });
 }});
```

Let me know if this makes sense

Reply:
updated! had to tweak sendEscrow also since it was relying on getBalance to setState for amount and loading